### PR TITLE
add package search on load if url parameters exist

### PR DESCRIPTION
**Pull request by classabbyamp**
I like to add custom search engines to my browser (vivaldi) for easy site-specific searches. Currently the package search on the void website ignores the url query string that get filled in when adding the package search as a search engine (see images below).

My changes allow for adding the package search as a search engine by running the query given in the query string if a valid architecture is given.

### Adding a search engine
Screenshot_20210603_230752

image

### The result
image

### CODE REVIEW

File: assets/js/voidsearch.js [hunk 1]
```diff
@@ -182,6 +200,13 @@
         );
     }
 
+    function getUrlParameter(name) {
```
Comment (ericonr) [inline assets/js/voidsearch.js:203]: aw, is there no standard function to do this? :/

Comment (ericonr) [reply assets/js/voidsearch.js:203]: Note that I don't actually know any JS.

Comment (Johnnynator) [reply assets/js/voidsearch.js:203]: https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams

Comment (classabbyamp) [reply assets/js/voidsearch.js:203]: There is a standard function that has support for all browsers but IE. Do we care about IE? If not, I can switch over to the built-in.

Comment (classabbyamp) [reply assets/js/voidsearch.js:203]: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Comment (ericonr) [reply assets/js/voidsearch.js:203]: I think it's fair to not care about IE here.

Comment (classabbyamp) [reply assets/js/voidsearch.js:203]: Ok, I'll swap it out.

**Comment by Johnnynator** [comment]
Wouldn't it be wise to also update the URL after a query? Right now the feature would be really hard to discover.

**Comment by classabbyamp** [comment]
oh yeah that's a good idea. I tried to implement switching the arch selector to the arch in the query and putting the query in the search box but couldn't figure it out.

**Comment by classabbyamp** [comment]
Alright, the package search should now run based on the url query parameters. It does this by having the form send a request to the same url instead of using a javascript event handler.

The form should validate the arch given in params and fill in the arch and query in the form based on the params.

**Comment by classabbyamp** [comment]
@ericonr or @Johnnynator, can I get a re-review on this?

**Review by the-maldridge — APPROVED** [review]
@ericonr @Johnnynator this looks okay to me, do you have any remaining concerns prior to merge?

**Review by Johnnynator — CHANGES_REQUESTED** [review]
You could have updated the url in (the old) `packageQuery(e)` with smth in the line of `history.replaceState(null, null, '?arch=' + archs.val() + '&q=' + q)` and not removed `form.submit(packageQuery);`
Your current solution has the problem, that it reloads the page on every search.

**Comment by classabbyamp** [comment]
hm would it be better to use `history.replaceState()` or [`history.pushState()`](https://developer.mozilla.org/en-US/docs/Web/API/History/pushState)?

**Comment by Johnnynator** [comment]
> hm would it be better to use `history.replaceState()` or [`history.pushState()`](https://developer.mozilla.org/en-US/docs/Web/API/History/pushState)?

Not sure, I probably would tend more to `replaceState` since I personally do not care (or want) a history of all my package searches, and I would prefer to be able to just get back to my previous site with just one click back in history.

**Comment by classabbyamp** [comment]
@Johnnynator it should work how you want it now.

**Comment by Johnnynator** [comment]
> @Johnnynator it should work how you want it now.

Yep, looks fine.
Thx for your work.

**Pull request merged**